src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/AssetOccupancyProviders/AssetOccupancyProviderBase.cs

```diff
@@ -371,15 +369,66 @@ public virtual Task HandleCanOccupyResultAsync(ICanOccupyResult result)
 
         return Task.CompletedTask;
     }
+    
+    protected virtual async Task<List<PeriodOccupancyModel>> InternalGetPeriodsAsync(Asset asset,
+        AssetCategory categoryOfAsset,
+        PeriodScheme periodScheme,
+        DateTime targetDate,
+        DateTime? currentDateTime = default)
+    {
+        var defaultAvailable = !asset.Disabled && !categoryOfAsset.Disabled &&
+                               await GetEffectivePeriodUsableAsync(asset, categoryOfAsset) is PeriodUsable.Accept;
+
+        var defaultAvailableVolume = defaultAvailable ? asset.Volume : 0;
+
+        var timeInAdvance = await GetEffectiveTimeInAdvanceAsync(asset, categoryOfAsset);
+
+        var models = periodScheme.Periods.Select(x => new PeriodOccupancyModel(targetDate, x.StartingTime,
+            x.GetEndingTime(), periodScheme.Id, x.Id, asset.Volume, defaultAvailableVolume)).ToList();
+
+        var schedules = await AssetScheduleRepository.GetListAsync(targetDate, asset.Id, periodScheme.Id);
+        var occupancies = await ProviderGetAssetOccupanciesAsync(targetDate, asset.Id);
+
+        var periodIdScheduleMapping = schedules.ToDictionary(x => x.PeriodId);
+        UpdatePeriodsUsableBySchedules(models, periodIdScheduleMapping);
+        UpdatePeriodsUsableByTimeInAdvances(models, periodIdScheduleMapping, timeInAdvance,
+            currentDateTime ?? Clock.Now);
+        UpdatePeriodsUsableByOccupancies(models, occupancies);
+        return models;
+    }
+
+    protected virtual string CalculateLockName(AssetCategory category, PeriodScheme periodScheme,
```

Comment:
I think it doesn't need `periodScheme` since we can support a "multi-effective-period-scheme" in the future, and the latter should share their lock.